### PR TITLE
Update "frequency" fetch logic

### PR DESCRIPTION
**Pull request by arun-madhavan-013**
Replace framerate fetch with best match - from DisplayInfo property instead of Framerate plugin because DisplayInfo is the mandatory and standard implementation requirement for SoC.

Supported possible framerate value from DisplayInfo.framerate property would be:
```
        "FramerateUnknown",
        "Framerate23976",
        "Framerate24",
        "Framerate25",
        "Framerate2997",
        "Framerate30",
        "Framerate47952",
        "Framerate48",
        "Framerate50",
        "Framerate5994",
        "Framerate60",
        "Framerate11988",
        "Framerate120",
        "Framerate144"
```
And get_frequency_from_displayinfo_framerate(in) would return the following for each:
```
test => in FramerateUnknown: out 60
test => in Framerate23976: out 23.976
test => in Framerate24: out 24
test => in Framerate25: out 25
test => in Framerate2997: out 29.97
test => in Framerate30: out 30
test => in Framerate47952: out 47.952
test => in Framerate48: out 48
test => in Framerate50: out 50
test => in Framerate5994: out 59.94
test => in Framerate60: out 60
test => in Framerate11988: out 119.88
test => in Framerate120: out 120
test => in Framerate144: out 144
```
Conformance Test:
```
d35@d35:~/Desktop/dab-compliance-suite$ dabSingleTest SystemSettingsGetConformance

testing system/settings/get   {} ... 
system/settings/get Latency, Expected: 800 ms, Actual: 746 ms

[ PASS ]
{
  "audioOutputMode": "Auto",
  "audioOutputSource": "HDMI",
  "audioVolume": 40,
  "cec": true,
  "hdrOutputMode": "AlwaysHdr",
  "language": "en-US",
  "lowLatencyMode": false,
  "matchContentFrameRate": "EnabledAlways",
  "memc": false,
  "mute": true,
  "outputResolution": {
    "frequency": 60.0,
    "height": 1080,
    "width": 1920
  },
  "pictureMode": "Standard",
  "status": 200,
  "textToSpeech": false,
  "videoInputSource": "Home"
}
```
DAB Adapter Logs:
```
d35@d35:~/Desktop/dab-works/dab-adapter-rs$ ./target/debug/dab-adapter -b $DIP -d $DIP --debug true
DAB<->RDK Adapter ("0.6.0" - "3e439da")
Using default values for app lifecycle timeouts.
youtube         - resume_launch_timeout_ms       =  3000ms.
youtube         - exit_to_destroy_timeout_ms     =  2500ms.
youtube         - cold_launch_timeout_ms         =  6000ms.
youtube         - exit_to_background_timeout_ms  =  2000ms.
RDK request: {"jsonrpc":"2.0","id":1,"method":"org.rdk.System.getDeviceInfo","params":{"params":["estb_mac"]}}
RDK response: {"jsonrpc":"2.0","id":1,"result":{"estb_mac":"02:AD:32:01:C8:35","success":true}}
DAB Device ID: 02AD3201C835
Ready to process DAB requests
processing: system/settings/get
RDK request: {"jsonrpc":"2.0","id":1,"method":"org.rdk.UserPreferences.1.getUILanguage"}
RDK response: {"jsonrpc":"2.0","id":1,"result":{"ui_language":"en-US","success":true}}
RDK request: {"jsonrpc":"2.0","id":1,"method":"DisplayInfo.width"}
RDK response: {"jsonrpc":"2.0","id":1,"result":1920}
RDK request: {"jsonrpc":"2.0","id":1,"method":"DisplayInfo.height"}
RDK response: {"jsonrpc":"2.0","id":1,"result":1080}
RDK request: {"jsonrpc":"2.0","id":1,"method":"DisplayInfo.framerate"}
RDK response: {"jsonrpc":"2.0","id":1,"result":"Framerate60"}
RDK request: {"jsonrpc":"2.0","id":2,"method":"org.rdk.DisplaySettings.getConnectedAudioPorts"}
RDK response: {"jsonrpc":"2.0","id":2,"result":{"connectedAudioPorts":["HDMI0","SPDIF0"],"success":true}}
RDK request: {"jsonrpc":"2.0","id":3,"method":"org.rdk.DisplaySettings.getVolumeLevel","params":{"audioPort":"HDMI0"}}
RDK response: {"jsonrpc":"2.0","id":3,"result":{"volumeLevel":"40.000000","success":true}}
RDK request: {"jsonrpc":"2.0","id":4,"method":"org.rdk.DisplaySettings.getConnectedAudioPorts"}
RDK response: {"jsonrpc":"2.0","id":4,"result":{"connectedAudioPorts":["HDMI0","SPDIF0"],"success":true}}
RDK request: {"jsonrpc":"2.0","id":5,"method":"org.rdk.DisplaySettings.getMuted","params":{"audioPort":"HDMI0"}}
RDK response: {"jsonrpc":"2.0","id":5,"result":{"muted":true,"success":true}}
RDK request: {"jsonrpc":"2.0","id":6,"method":"org.rdk.HdmiCec_2.getEnabled"}
RDK response: {"jsonrpc":"2.0","id":6,"result":{"enabled":true,"success":true}}
RDK request: {"jsonrpc":"2.0","id":7,"method":"org.rdk.DisplaySettings.getSettopHDRSupport"}
RDK response: {"jsonrpc":"2.0","id":7,"result":{"supportsHDR":true,"standards":["HDR10","HLG","Dolby Vision"],"success":true}}
RDK request: {"jsonrpc":"2.0","id":8,"method":"org.rdk.DisplaySettings.getTvHDRSupport"}
RDK response: {"jsonrpc":"2.0","id":8,"result":{"supportsHDR":true,"standards":["HDR10","HLG","Dolby Vision"],"success":true}}
RDK request: {"jsonrpc":"2.0","id":9,"method":"org.rdk.DisplaySettings.getConnectedAudioPorts"}
RDK response: {"jsonrpc":"2.0","id":9,"result":{"connectedAudioPorts":["HDMI0","SPDIF0"],"success":true}}
RDK request: {"jsonrpc":"2.0","id":10,"method":"org.rdk.DisplaySettings.getSoundMode","params":{"audioPort":"HDMI0"}}
RDK response: {"jsonrpc":"2.0","id":10,"result":{"soundMode":"AUTO (Dolby Digital Plus)","success":true}}
RDK request: {"jsonrpc":"2.0","id":11,"method":"org.rdk.DisplaySettings.getConnectedAudioPorts"}
RDK response: {"jsonrpc":"2.0","id":11,"result":{"connectedAudioPorts":["HDMI0","SPDIF0"],"success":true}}
RDK request: {"jsonrpc":"2.0","id":12,"method":"org.rdk.TextToSpeech.isttsenabled"}
RDK response: {"jsonrpc":"2.0","id":12,"result":{"isenabled":false,"TTS_Status":0,"success":true}}
Publishing response: dab/_response/system/settings/get {"audioOutputMode":"Auto","audioOutputSource":"HDMI","audioVolume":40,"cec":true,"hdrOutputMode":"AlwaysHdr","language":"en-US","lowLatencyMode":false,"matchContentFrameRate":"EnabledAlways","memc":false,"mute":true,"outputResolution":{"frequency":60.0,"height":1080,"width":1920},"pictureMode":"Standard","status":200,"textToSpeech":false,"videoInputSource":"Home"}
```

### CODE REVIEW

File: src/device/rdk/interface.rs [hunk 1]
```diff
@@ -579,6 +579,19 @@ pub fn get_thunder_property(method_name: &str, key_name: &str) -> Result<String,
     }
 }
 
+pub fn get_frequency_from_displayinfo_framerate(framerate: &str) -> Result<f32, std::num::ParseFloatError> {
```
Comment (xhaakon) [inline src/device/rdk/interface.rs:582]: There's just a single place from where `get_frequency_from_displayinfo_framerate()` is called. Unless you have a clear intention to reuse this function in a code you're about to push in a near future, I'd rather move this couple of lines inside `get_rdk_resolution()`.

No need to pollute `interface.rs` with "functions" that aren't actually used as such.

File: src/device/rdk/system/settings/get.rs [hunk 2]
```diff
@@ -28,31 +27,19 @@ fn get_rdk_language() -> Result<String, DabError> {
...
+    let displayinfo_height = get_thunder_property("DisplayInfo.height", "")?;
+    let height = displayinfo_height.parse::<u32>().map_err(|_| DabError::Err400("Invalid height(parse to u32 failed)".to_string()))?;
 
-    let mut dimensions = rdkresponse
-        .result
-        .framerate
-        .trim_end_matches(']')
-        .split('x');
+    let displayinfo_framerate = get_thunder_property("DisplayInfo.framerate", "")?;
+    let frequency = get_frequency_from_displayinfo_framerate(&displayinfo_framerate).unwrap_or(60.0);
```
Comment (xhaakon) [inline src/device/rdk/system/settings/get.rs:37]: Parsing of frequency currently doesn't work correctly due to an issue in `get_thunder_property()`. We have to first merge [this fix](https://github.com/device-automation-bus/dab-adapter-rs/pull/58).

**Review by xhaakon — CHANGES_REQUESTED** [review]

File: src/device/rdk/interface.rs (hunk 1 above)
Comment (arun-madhavan-013) [reply src/device/rdk/interface.rs]: Thank you for the fix logic; pulled same into this to avoid merge conflict.

**Comment by arun-madhavan-013** [comment]
Raised https://github.com/device-automation-bus/dab-adapter-rs/pull/59

**Pull request closed without merging**